core/amber/src/main/scala/edu/uci/ics/texera/web/resource/aiassistant/AiAssistantResource.scala

```diff
@@ -39,6 +38,60 @@ class AIAssistantResource {
   @Path("/isenabled")
   def isAIAssistantEnable: String = isEnabled
 
+  /**
+    * Endpoint to get the operator comment from OpenAI.
+    * @param prompt The input prompt for the OpenAI model.
+    * @param user The authenticated session user.
+    * @return A response containing the generated comment from OpenAI or an error message.
+    */
+  @POST
+  @Path("/openai")
+  @Consumes(Array(MediaType.APPLICATION_JSON))
+  def getAiComment(prompt: String, @Auth user: SessionUser): Response = {
+    // Prepare the final prompt by escaping necessary characters
+    val finalPrompt = prompt.replace("\\", "\\\\").replace("\"", "\\\"")
+
+    // Create the JSON request body
+    val requestBody =
+      s"""
+         |{
+         |  "model": "gpt-4o",
+         |  "messages": [{"role": "user", "content": "$finalPrompt"}],
+         |  "max_tokens": 1000
+         |}
+       """.stripMargin
+
+    try {
+      // Set up the connection to the OpenAI API
+      val url = new java.net.URL("https://api.openai.com/v1/chat/completions")
```

Comment:
Try to mimic the `getAnnotation` way of sending the http requests, i.e. `Unirest.post`